src/lib/statistics/statistics_objects/min_max_filter.cpp

```diff
@@ -158,6 +158,49 @@ bool MinMaxFilter<T>::does_not_contain(const PredicateCondition predicate_condit
       const auto value2 = boost::get<T>(*variant_value2);
       return value >= max || value2 <= min;
     }
+    case PredicateCondition::NotLike:
+    case PredicateCondition::Like: {
+      if constexpr (std::is_same_v<T, pmr_string>) {
+        const auto pattern = boost::get<pmr_string>(variant_value);
+
+        const auto single_char_wildcard_pos = pattern.find_first_of('_');
+        const auto multi_char_wildcard_pos = pattern.find_first_of('%');
+
+        const auto single_valid_char_wildcard_exist = (single_char_wildcard_pos != std::string::npos);
+        const auto multi_char_wildcard_pos_exsists = (multi_char_wildcard_pos != std::string::npos);
+
+        auto wildcard_pos = size_t{};
+
+        if (!single_valid_char_wildcard_exist && !multi_char_wildcard_pos_exsists) {
+          return false;
+        }
+        if (single_char_wildcard_pos == 0 || multi_char_wildcard_pos == 0) {
+          return false;
+        }
+        if (single_valid_char_wildcard_exist && single_char_wildcard_pos != 0) {
+          wildcard_pos = single_char_wildcard_pos;
+        }
+        if (multi_char_wildcard_pos_exsists && multi_char_wildcard_pos != 0 && wildcard_pos < multi_char_wildcard_pos) {
+          wildcard_pos = multi_char_wildcard_pos;
+        }
+
+        // Calculate lower bound of the search pattern
+        const auto lower_bound = pattern.substr(0, wildcard_pos);
+        const auto current_character_value = lower_bound.back();
```

Comment:
What is "current" about this? Also what does `value` convey here?

Reply:
I renamed it to `last_character_of_lower_bound`. I hope the readability is now better.